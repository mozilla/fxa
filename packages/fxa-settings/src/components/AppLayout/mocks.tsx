/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { RelierCmsInfo } from '../../models/integrations';

export const MOCK_CMS_INFO_VALID_LINEAR_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_VALID_RADIAL_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'radial-gradient(circle, #ff6b6b, #4ecdc4)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_NO_BG = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_INVALID_BG_COLOR = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    backgrounds: {
      defaultLayout: 'solid-color-red',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_WITH_PAGE_TITLE = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
    pageTitle: 'CMS Custom Title',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_NO_PAGE_TITLE = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/logo.png',
    logoAltText: 'Test App Logo',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: undefined,
    logoUrl: undefined,
    logoAltText: undefined,
    headerLogoUrl: 'https://example.com/cms-logo.png',
    headerLogoAltText: 'CMS Custom Logo',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO_NO_ALT = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: undefined,
    logoUrl: undefined,
    logoAltText: undefined,
    headerLogoUrl: 'https://example.com/cms-logo.png',
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_DEFAULT_LOGO = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    buttonColor: '#0078d4',
    logoUrl: undefined,
    logoAltText: undefined,
  },
} as RelierCmsInfo;

export const MOCK_CMS_INFO_HEADER_LOGO_WITH_OTHER_PROPS = {
  name: 'Test App',
  clientId: 'test123',
  entrypoint: 'test',
  shared: {
    headerLogoUrl: 'https://example.com/cms-logo.png',
    headerLogoAltText: 'CMS Custom Logo',
    buttonColor: '#0078d4',
    logoUrl: 'https://example.com/other-logo.png',
    logoAltText: 'Other Logo',
    backgrounds: {
      defaultLayout: 'linear-gradient(135deg, #667eea 0%, #764ba2 100%)',
    },
    pageTitle: 'Test App - Custom Title',
  },
} as RelierCmsInfo;

export const INTEGRATION_WITH_UNDEFINED_GETCMSINFO = {
  getCmsInfo: () => undefined,
};

export const INTEGRATION_WITH_NO_GETCMSINFO = {} as Record<string, never>;

export function createIntegration(cmsInfo?: RelierCmsInfo) {
  return {
    getCmsInfo: () => cmsInfo,
  };
}
