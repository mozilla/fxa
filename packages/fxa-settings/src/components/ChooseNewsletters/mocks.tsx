/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { useState } from 'react';
import ChooseNewsletters from '.';
import { newsletters } from './newsletters';

export const SubjectWithNewsletters = () => {
  const [, setSelected] = useState<string[]>([]);

  const newsletters = [
    {
      label: 'Early access to test new products',
      slug: ['test-pilot'],
      ftlId: 'choose-newsletters-option-test-pilot',
    },
  ];
  return (
    <ChooseNewsletters
      {...{ newsletters }}
      setSelectedNewsletterSlugs={setSelected}
    />
  );
};

export const SubjectWithNone = () => {
  const [, setSelected] = useState<string[]>([]);

  return (
    <ChooseNewsletters
      {...{ newsletters }}
      setSelectedNewsletterSlugs={setSelected}
    />
  );
};
