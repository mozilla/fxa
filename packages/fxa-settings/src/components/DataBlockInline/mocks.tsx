/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import DataBlockInline, { DataBlockInlineProps } from '.';

export const Subject = ({
  value = 'ANMD 1S09 7Y2Y 4EES 02CW BJ6Z PYKP H69F',
  ...props // overrides
}: Partial<DataBlockInlineProps> = {}) => (
  <DataBlockInline {...{ value, ...props }} />
);
