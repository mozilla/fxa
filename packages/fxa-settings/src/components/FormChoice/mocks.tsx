/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { CHOICES } from '.';
import {
  BackupAuthenticationCodesImage,
  BackupRecoveryPhoneSmsImage,
} from '../images';

export const commonFormChoiceProps = {
  isSubmitting: false,
  onSubmit: (data: any) => console.log('Submitted with choice:', data),
  legendEl: (
    <legend>
      <h2 className="card-header mb-4">I am (form) legend</h2>
    </legend>
  ),
  formChoices: [
    {
      id: 'recovery-choice-phone',
      value: CHOICES.phone,
      image: <BackupRecoveryPhoneSmsImage ariaHidden />,
      localizedChoiceTitle: '1st Choice Example',
      localizedChoiceInfo: '1st example description',
    },
    {
      id: 'recovery-choice-code',
      value: CHOICES.code,
      image: <BackupAuthenticationCodesImage ariaHidden />,
      localizedChoiceTitle: '2nd Choice Example',
      localizedChoiceInfo: '2nd example description',
    },
  ],
};

export const commonFormChoicePropsWithCms = {
  ...commonFormChoiceProps,
  cmsButton: {
    color: '#592ACB',
    text: 'Continue with CMS',
  },
};
