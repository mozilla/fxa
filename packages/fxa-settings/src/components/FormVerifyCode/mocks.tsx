/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import FormVerifyCode, { FormAttributes } from '.';
import { useFtlMsgResolver } from '../../models';

export const MOCK_EMAIL = 'test@example.com';

export const Subject = () => {
  type FormData = {
    code?: string;
  };

  const [code, setCode] = useState<string>('');
  const [codeErrorMessage, setCodeErrorMessage] = useState<string>('');
  const ftlMsgResolver = useFtlMsgResolver();

  const formAttributes: FormAttributes = {
    inputFtlId: 'demo-input-label-id',
    inputLabelText: 'Enter your 4-digit code',
    pattern: '[0-9]{4}',
    maxLength: 4,
    submitButtonFtlId: 'demo-submit-button-id',
    submitButtonText: 'Check that code',
  };

  const onFormSubmit = () => {
    if (!code) {
      const codeRequiredError = ftlMsgResolver.getMsg(
        'confirm-signup-code-required-error',
        'Confirmation code required'
      );
      setCodeErrorMessage(codeRequiredError);
    }
  };

  const { handleSubmit } = useForm<FormData>({
    mode: 'onBlur',
    criteriaMode: 'all',
    defaultValues: {
      code: '',
    },
  });

  return (
    <FormVerifyCode
      email={MOCK_EMAIL}
      onSubmit={handleSubmit(onFormSubmit)}
      viewName="default-view"
      {...{
        formAttributes,
        code,
        setCode,
        codeErrorMessage,
        setCodeErrorMessage,
      }}
    />
  );
};
