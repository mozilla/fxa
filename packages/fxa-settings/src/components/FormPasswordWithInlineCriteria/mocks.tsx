/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import { useForm } from 'react-hook-form';
import FormPasswordWithInlineCriteria, { PasswordFormType } from '.';
import { MOCK_ACCOUNT } from '../../models/mocks';

type SubjectProps = {
  passwordFormType: PasswordFormType;
  requirePasswordConfirmation?: boolean;
};

export const Subject = ({
  passwordFormType,
  requirePasswordConfirmation,
}: SubjectProps) => {
  type FormData = {
    oldPassword?: string;
    newPassword: string;
    confirmPassword: string;
  };
  const onFormSubmit = () => {
    // this alert is for Storybook
    alert('Form submitted! (onFormSubmit called)');
  };

  const { handleSubmit, register, getValues, errors, formState, trigger } =
    useForm<FormData>({
      mode: 'onTouched',
      criteriaMode: 'all',
      defaultValues: {
        newPassword: '',
        confirmPassword: '',
      },
    });

  return (
    <FormPasswordWithInlineCriteria
      {...{
        formState,
        errors,
        trigger,
        register,
        getValues,
        passwordFormType,
        requirePasswordConfirmation,
      }}
      onSubmit={handleSubmit(onFormSubmit)}
      email={MOCK_ACCOUNT.primaryEmail.email}
      loading={false}
      onFocusMetricsEvent={() => {}}
    />
  );
};
