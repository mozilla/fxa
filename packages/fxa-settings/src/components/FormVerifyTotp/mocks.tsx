/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React, { useCallback, useState } from 'react';
import AppLayout from '../AppLayout';
import FormVerifyTotp from '.';
import { FormVerifyTotpProps } from './interfaces';

export const Subject = ({
  codeLength = 6,
  codeType = 'numeric',
  success = true,
  initialErrorMessage = '',
}: Partial<FormVerifyTotpProps> & {
  success?: Boolean;
  initialErrorMessage?: string;
}) => {
  const localizedInputLabel = `Enter ${codeLength.toString()}-digit code`;
  const localizedSubmitButtonText = 'Submit';
  const [errorMessage, setErrorMessage] = useState(initialErrorMessage);

  const mockVerifyCodeSuccess = useCallback(
    (code: string) => alert(`Mock code submission with code ${code}`),
    []
  );

  const mockVerifyCodeFail = useCallback(
    (code: string) => setErrorMessage('Something went wrong'),
    []
  );

  const verifyCode = success ? mockVerifyCodeSuccess : mockVerifyCodeFail;

  return (
    <AppLayout>
      <FormVerifyTotp
        {...{
          codeLength,
          codeType,
          errorMessage,
          localizedInputLabel,
          localizedSubmitButtonText,
          setErrorMessage,
          verifyCode,
        }}
      />
    </AppLayout>
  );
};

export default Subject;
