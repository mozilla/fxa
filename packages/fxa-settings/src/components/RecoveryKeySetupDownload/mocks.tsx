/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import RecoveryKeySetupDownload from '.';
import { MOCK_ACCOUNT } from '../../models/mocks';
import { MOCK_RECOVERY_KEY } from '../../pages/mocks';

export const Subject = () => {
  return (
    <RecoveryKeySetupDownload
      recoveryKeyValue={MOCK_RECOVERY_KEY}
      email={MOCK_ACCOUNT.primaryEmail.email}
      navigateForward={() => Promise.resolve()}
      viewName="viewName"
    />
  );
};
