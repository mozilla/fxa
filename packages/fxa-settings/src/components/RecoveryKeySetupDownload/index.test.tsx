/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { screen, waitFor, within } from '@testing-library/react';
import { renderWithLocalizationProvider } from 'fxa-react/lib/test-utils/localizationProvider';
import { Subject } from './mocks';
import { MOCK_RECOVERY_KEY } from '../../pages/mocks';

describe('RecoveryKeySetupDownload', () => {
  it('renders as expected', async () => {
    renderWithLocalizationProvider(<Subject />);

    screen.getByText(MOCK_RECOVERY_KEY);
    screen.getByRole('button', { name: 'Copy' });

    screen.getByRole('heading', {
      level: 3,
      name: 'Places to store your key:',
    });
    const list = screen.getByRole('list', {
      name: 'Places to store your key:',
    });
    const listItems = within(list).getAllByRole('listitem');
    expect(listItems.length).toBe(4);

    await waitFor(
      () => {
        const b = screen.getByRole('button', { name: 'Download and continue' });
        expect(b).toBeInTheDocument();
      },
      { timeout: 2000 }
    );

    screen.getByRole('button', {
      name: 'Continue without downloading',
    });
  });
});
