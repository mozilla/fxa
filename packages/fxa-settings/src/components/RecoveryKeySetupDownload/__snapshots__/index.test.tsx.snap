// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecoveryKeySetupDownload renders button with CMS passthrough 1`] = `
<button
  class="cta-primary-cms cta-xl w-full mt-4"
  style="--cta-bg: #D41C1C; --cta-border: #D41C1C; --cta-active: #D41C1C; --cta-disabled: #D41C1C60;"
>
  Download and continue
</button>
`;
